File: tks/ApplicationCore/DomainModel/src/test/kotlin/model/values/EmailTest.java
package model.values;

import exceptions.TypeValidationFailedException;
import lombok.SneakyThrows;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class EmailTest {

    @SneakyThrows
    @Test
    public void equalityTest(){
        var email1 = new Email("mszewc@edu.pl");
        var email2 = new Email("mszewc@edu.pl");
        var email3 = new Email("mzab@edu.pl");

        assertNotSame(email1, email2);
        assertEquals(email1, email2);
        assertNotEquals(email1, email3);
    }

    @Test
    public void constructorValidationTest(){
        assertDoesNotThrow(() -> new Email("mszewc@edu.pl"));

        assertThrows(TypeValidationFailedException.class, () -> new Email("mszewc"));

        assertThrows(TypeValidationFailedException.class, () -> new Email("mszewc@edu.pl@hehe"));

        assertThrows(TypeValidationFailedException.class, () -> new Email(null));
        assertThrows(TypeValidationFailedException.class, () -> new Email(""));
    }

}

File: tks/ApplicationCore/DomainModel/src/test/kotlin/domain/model/values/EmailTest.kt
package domain.model.values

import domain.exceptions.TypeValidationFailedException
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

internal class EmailTest {

    @Test
    fun equalityTest() {
        val email1 = Email("mszewc@edu.pl")
        val email2 = Email("mszewc@edu.pl")
        val email3 = Email("mzab@edu.pl")
        Assertions.assertNotSame(email1, email2)
        Assertions.assertEquals(email1, email2)
        Assertions.assertNotEquals(email1, email3)
    }

    @Test
    fun constructorValidationTest() {
        Assertions.assertDoesNotThrow<Email> { Email("mszewc@edu.pl") }
        Assertions.assertThrows(TypeValidationFailedException::class.java) { Email("mszewc") }
        Assertions.assertThrows(TypeValidationFailedException::class.java) { Email("mszewc@edu.pl@hehe") }
        Assertions.assertThrows(TypeValidationFailedException::class.java) { Email("") }
    }
}